sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -111,8 +110,11 @@ class Column(protected[sql] val expr: Expression) extends Logging {
     // Leave an unaliased generator with an empty list of names since the analyzer will generate
     // the correct defaults after the nested expression's type has been resolved.
     case explode: Explode => MultiAlias(explode, Nil)
+
     case jt: JsonTuple => MultiAlias(jt, Nil)
 
+    case func: UnresolvedFunction => UnresolvedAlias(func)
```

Comment:
I missed one thing, this will change the alias of functions which are not generators. Maybe we should add an optionName in `UnresolvedAlias` as the default aliasing name instead of `c0`, `c1`, etc.